config_parser.cc

```diff
@@ -199,12 +201,18 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
           new_config);
       config_stack.push(new_config);
     } else if (token_type == TOKEN_TYPE_END_BLOCK) {
-      if (last_token_type != TOKEN_TYPE_STATEMENT_END) {
+      nested--;
+      if (! (last_token_type == TOKEN_TYPE_STATEMENT_END 
```

Comment:
Slightly inconsistent coding style. Rest of class uses infix != instead of !( ... == ... ). But not important.